libraries/ms-common/src/stm32f0xx/pwm.c

```diff
@@ -28,23 +78,25 @@ StatusCode pwm_init(uint16_t period_ms) {
     .TIM_RepetitionCounter = 0,
   };
 
-  TIM_TimeBaseInit(TIM3, &tim_init);
-  TIM_Cmd(TIM3, ENABLE);
-  TIM_CtrlPWMOutputs(TIM3, ENABLE);
+  TIM_TimeBaseInit(s_timer_def[timer], &tim_init);
+  TIM_Cmd(s_timer_def[timer], ENABLE);
+  TIM_CtrlPWMOutputs(s_timer_def[timer], ENABLE);
 
   return STATUS_CODE_OK;
 }
 
-uint16_t pwm_get_period(void) {
-  return s_period_ms;
+uint16_t pwm_get_period(PWMTimer timer) {
+  status_ok_or_return(prv_check_timer_id(timer));
+  return s_period_ms[timer];
 }
 
-StatusCode pwm_set_pulse(uint16_t pulse_width_ms) {
-  if (s_period_ms == 0) {
+StatusCode pwm_set_pulse(PWMTimer timer, uint16_t pulse_width_ms) {
+  if (s_period_ms[timer] == 0) {
     return status_msg(STATUS_CODE_UNINITIALIZED, "Pwm must be initialized.");
-  } else if (pulse_width_ms > s_period_ms) {
+  } else if (pulse_width_ms > s_period_ms[timer]) {
     return status_msg(STATUS_CODE_INVALID_ARGS, "Pulse width must be leq period.");
   }
+  status_ok_or_return(prv_check_timer_id(timer));
```

Comment:
I'm not a huge fan of these "check and return status" functions because the status code will be from the check function and not this one. Since we have no way of storing the caller, the error code isn't very helpful at that point.